great_expectations/data_context/types/base.py

```diff
@@ -2016,6 +2017,80 @@ def __init__(
 
         super().__init__(commented_map=commented_map)
 
+    def to_json_dict(self) -> dict:
```

Comment:
@NathanFarmer something does not feel right about putting `to_json_dict()` here and using the very general `super()`.  Moreover, is `batch_request` in this class (`CheckpointConfig`) *ever* anything other than a `dict`?  Should not be, because configuration admits only the basic objects, plus lists and dictionaries, but not complex typed objects.  However, I may be missing a point here, so happy to discuss.  Thank you.

Reply:
`batch_request` can be passed into the `Checkpoint` as a `BatchRequest` object (as opposed to yml config). The `BatchRequest` is then passed into `CheckpointConfig` and returned without ever being converted to `dict`. The type hint for `CheckpointConfig` does indicate that a `dict` is expected, but this can be changed (or not depending on the route we take). The use of `super()` here for `LegacyCheckpoint` can be addressed by handling each parameter in `LegacyCheckpoint` if that is considered best practice.

Reply:
@alexsherstinsky I have removed this method